### PR TITLE
Feature/login formik yup

### PR DESCRIPTION
**Pull request by pink-bunny**
- expand linter rules;
- create button and input components;
- connect formik;
- add yup validation;
- dispatch action on submit login form.

### CODE REVIEW

File: src/components/InputField/component.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  form: { touched, errors },
+  ...props
+}) => {
+  const fieldTouched = getIn(touched, field.name);
+  const fieldErrors = getIn(errors, field.name);
+
+  return (
+    <Form.Item
+      validateStatus={fieldErrors && fieldTouched && 'error'}
+      help={fieldErrors && fieldTouched && fieldErrors}
```
Comment (plzen) [inline src/components/InputField/component.js]: Form.Item accepts string as text message, but you might have `false` in this construction

Comment (plzen) [inline src/components/InputField/component.js]: You have duplicated `fieldErrors` here

File: src/components/InputField/component.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        placeholder={placeholder}
+        prefix={(
+          icon && (
+            <Icon
+              type={icon}
+              style={{ color: 'rgba(0,0,0,.25)' }}
+            />
+          )
+        )}
+        {...{ ...field, ...props }}
```
Comment (plzen) [inline src/components/InputField/component.js]: Just use construction without wrapping it with additional object, like
```
{...field}
{...props}
```

This is more readable

Comment (pink-bunny) [reply src/components/InputField/component.js]: Done.

File: src/components/InputField/component.js (hunk 1 above)
Comment (pink-bunny) [reply src/components/InputField/component.js]: In first-time fieldErrors works as condition (if it returns string than fieldErrors=true).
The second time it is message.

help={fieldErrors && fieldTouched && fieldErrors}
equals
help={fieldErrors && fieldTouched ? fieldErrors : ''}

Comment (pink-bunny) [reply src/components/InputField/component.js]: The second fieldErrors gives an error message to the help field.

Comment (pink-bunny) [reply src/components/InputField/component.js]: Fixed.

Comment (pink-bunny) [reply src/components/InputField/component.js]: Fixed.

**Pull request merged**